dali/pipeline/operators/reader/coco_reader_op.cc

```diff
@@ -14,47 +14,99 @@
+  .AddOptionalArg(
+    "meta_files_path",
+    "Path to directory with meta files containing preprocessed COCO annotations.",
+    std::string())
+  .AddOptionalArg("annotations_file",
+      R"code(List of paths to the JSON annotations files.)code",
+      std::string())
+  .AddOptionalArg("shuffle_after_epoch",
+      R"code(If true, reader shuffles whole dataset after each epoch.)code",
+      false)
   .AddArg("file_root",
       R"code(Path to a directory containing data files.)code",
       DALI_STRING)
-  .AddArg("annotations_file",
-      R"code(List of paths to the JSON annotations files.)code",
-      DALI_STRING_VEC)
-  .AddOptionalArg("file_list",
-      R"code(Path to the file with a list of pairs ``file label``
-(leave empty to traverse the `file_root` directory to obtain files and labels))code",
-      std::string())
   .AddOptionalArg("ltrb",
       R"code(If true, bboxes are returned as [left, top, right, bottom], else [x, y, width, height].)code",
       false)
-  .AddOptionalArg("ratio",
-      R"code(If true, bboxes returned values as expressed as ratio w.r.t. to the image width and height.)code",
+  .AddOptionalArg("skip_empty",
+      R"code(If true, reader will skip samples with no object instances in them)code",
       false)
   .AddOptionalArg("size_threshold",
       R"code(If width or height of a bounding box representing an instance of an object is under this value,
 object will be skipped during reading. It is represented as absolute value.)code",
       0.1f,
       false)
-  .AddOptionalArg("skip_empty",
-      R"code(If true, reader will skip samples with no object instances in them)code",
+  .AddOptionalArg("ratio",
+      R"code(If true, bboxes returned values as expressed as ratio w.r.t. to the image width and height.)code",
       false)
+  .AddOptionalArg("file_list",
+      R"code(Path to the file with a list of pairs ``file id``
+(leave empty to traverse the `file_root` directory to obtain files and labels))code",
+      std::string())
   .AddOptionalArg("save_img_ids",
       R"code(If true, image IDs will also be returned.)code",
       false)
-  .AddOptionalArg("shuffle_after_epoch",
-      R"code(If true, reader shuffles whole dataset after each epoch.)code",
+  .AddOptionalArg("dump_meta_files",
+      R"code(If true, operator will dump meta files in folder provided with `dump_meta_files_path`.)code",
       false)
+  .AddOptionalArg(
+    "dump_meta_files_path",
+    "Path to directory for saving meta files containing preprocessed COCO annotations.",
+    std::string())
   .AdditionalOutputsFn([](const OpSpec& spec) {
     return static_cast<int>(spec.GetArgument<bool>("save_img_ids"));
   })
   .AddParent("LoaderBase");
+
+
+void COCOReader::ValidateOptions(const OpSpec &spec) {
+  DALI_ENFORCE(
+    spec.HasArgument("meta_files_path") || spec.HasArgument("annotations_file"),
+    "`meta_files_path` or `annotations_file` must be provided");
+  DALI_ENFORCE(
+    !spec.HasArgument("file_list"),
```

Comment:
Right now it has to be there, because we pass spec all the way to `FileLoader` and it checks this argument, so it's better to give verbose error message from the `COCOReader`